supervisor/api/ingress.py

```diff
@@ -99,21 +125,26 @@ async def handler(
         # Process requests
         addon = self._extract_addon(request)
         path = request.match_info.get("path")
+        session_data = self.sys_ingress.sessions_data[session]
```

Comment:
that is wrong, this only return a dict. I think you should create an ``sys_ingress.get_data(session_id: str) -> SessionIngressData` and trow and NotFound exception (there are some others at exception.py).

Reply:
@pvizeli You suggested (far) above, that the tests should be compatible with older versions of home assistant where the user_id is not sent with session creation https://github.com/home-assistant/supervisor/pull/4152#discussion_r1124163083.
Because in that case session_data is `None`, I left this as it is and changed the signature of `_handle_websocket/_handle_request` to expect a `None` IngressSessionData.

Reply:
That is fine, we can suppress the error until we force it